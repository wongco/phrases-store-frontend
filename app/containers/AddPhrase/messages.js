/*
 * AddPhrase Messages
 *
 * This contains all the text for the AddPhrase container.
 */

import { defineMessages } from 'react-intl';

export const scope = 'app.containers.AddPhrase';

export default defineMessages({
  header: {
    id: `${scope}.header`,
    defaultMessage: 'Add a new phrase!',
  },
});
