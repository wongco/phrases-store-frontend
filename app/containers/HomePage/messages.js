/*
 * HomePage Messages
 *
 * This contains all the text for the HomePage container.
 */
import { defineMessages } from 'react-intl';

export const scope = 'app.containers.HomePage';

export default defineMessages({
  header: {
    id: `${scope}.header`,
    defaultMessage: 'Welcome to The Phrases Store!',
  },
  intro: {
    id: `${scope}.intro`,
    defaultMessage: 'Click Add on the navigation to add a phrase.',
  },
});
