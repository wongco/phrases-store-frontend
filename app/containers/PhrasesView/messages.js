/*
 * PhrasesView Messages
 *
 * This contains all the text for the PhrasesView container.
 */

import { defineMessages } from 'react-intl';

export const scope = 'app.containers.PhrasesView';

export default defineMessages({
  header: {
    id: `${scope}.header`,
    defaultMessage: 'Ordered by most recent:',
  },
  error: {
    id: `${scope}.error`,
    defaultMessage: 'Error obtaining latest phrases!',
  },
  notice: {
    id: `${scope}.notice`,
    defaultMessage: 'No phrases added yet! Please add one!',
  },
});
