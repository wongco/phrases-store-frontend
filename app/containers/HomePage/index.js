/*
 * HomePage
 *
 * This is the first thing users see of our App, at the '/' route
 *
 * NOTE: while this component should technically be a stateless functional
 * component (SFC), hot reloading does not currently support SFCs. If hot
 * reloading is not a necessity for you then you can refactor it and remove
 * the linting exception.
 */

import React from 'react';
import { FormattedMessage } from 'react-intl';
import styled from 'styled-components';
import H1 from 'components/H1';
import messages from './messages';

const P = styled.p`
  font-family: Helvetica;
  font-weight: 200;
  text-align: center;
`;

/* eslint-disable react/prefer-stateless-function */
export default class HomePage extends React.PureComponent {
  render() {
    return (
      <div>
        <H1>
          <FormattedMessage {...messages.header} />
        </H1>
        <P>
          <FormattedMessage {...messages.intro} />
        </P>
      </div>
    );
  }
}
