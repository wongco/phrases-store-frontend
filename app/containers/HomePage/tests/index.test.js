import React from 'react';
import { FormattedMessage } from 'react-intl';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import HomePage from '../index';
import messages from '../messages';

describe('<HomePage />', () => {
  it('renders without crashing', () => {
    shallow(<HomePage />);
  });

  it('matches snapshot for default criteria', () => {
    const wrapper = shallow(<HomePage />);
    const serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
  });

  it('should render the page message', () => {
    const renderedComponent = shallow(<HomePage />);
    expect(
      renderedComponent.contains(<FormattedMessage {...messages.header} />),
    ).toEqual(true);
  });
});
