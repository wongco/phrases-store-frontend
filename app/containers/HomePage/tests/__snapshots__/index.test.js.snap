// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> matches snapshot for default criteria 1`] = `
<div>
  <H1>
    <FormattedMessage
      defaultMessage="Welcome to The Phrases Store!"
      id="app.containers.HomePage.header"
      values={Object {}}
    />
  </H1>
  <HomePage__P>
    <FormattedMessage
      defaultMessage="Click Add on the navigation to add a phrase."
      id="app.containers.HomePage.intro"
      values={Object {}}
    />
  </HomePage__P>
</div>
`;
